components/public-api-server/pkg/apiv1/oidc.go

```diff
@@ -334,17 +330,32 @@ func (s *OIDCService) SetClientConfigActivation(ctx context.Context, req *connec
 
 	config, err := db.GetOIDCClientConfig(ctx, s.dbConn, clientConfigID)
 	if err != nil {
+		if errors.Is(err, db.ErrorNotFound) {
+			return nil, connect.NewError(connect.CodeNotFound, fmt.Errorf("OIDC Client Config %s for Organization %s does not exist", clientConfigID.String(), organizationID.String()))
+		}
+
 		return nil, err
 	}
-	if config.Verified == nil || !*config.Verified {
-		log.Extract(ctx).WithError(err).Error("Failed to activate an unverified OIDC Client Config.")
-		return nil, connect.NewError(connect.CodeFailedPrecondition, fmt.Errorf("Failed to activate an unverified OIDC Client Config %s for Organization %s", clientConfigID.String(), organizationID.String()))
+
+	if req.Msg.Activate {
+		if config.Verified == nil || !*config.Verified {
```

Comment:
It feels odd that we'd need to do a `nil` check against a boolean. Do you know why the Verified field is a `*bool` in the DB model? I suspect we'd default to false anyway so we could change it to `bool`

Reply:
Reason for `*bool` instead of `bool` was the `Updates(update)` call on gorm in this [PR](https://github.com/gitpod-io/gitpod/pull/17576/files#diff-f04090b6374d14e0472814aa814febc4a4b1cda2f1b0e3f0581e745765503f71R243) which, as asked on slack, didn't work as expected, i.e. didn't reset the `Verified` field to `false` if it was set to `true` before. 

If you know how to reset the `Verified` in gorm's `Updates` param, that would be great. I got you on this like: either go with a unstructured map, or use `*bool`, both of which aren't great.

Reply:
Yep, those aren't great. Falling back to direct SQL would also be an option. I'm fine with this, didn't connect the reasoning in the first place.